docs/unmanaged_configuration.md

```diff
@@ -25,3 +25,39 @@ oc set env dc/fluentd MERGE_JSON_LOG=true
 * Possible log loss due to Elasticsearch rejecting documents due to inconsistent type mappings
 * Potential buffer storage leak caused by rejected message cycling
 * Overwrite of data for field with same names
+
+## ElasticSearch
+### Exposing elasticsearch service with a route
+```
+# Get ElasticSearch pod name
+ESPOD=$( oc get pods -l component=elasticsearch -o name | sed -e "s/pod\///" )
+
+# Extract CA cert.
+oc exec $ESPOD -- cat /etc/openshift/elasticsearch/secret/admin-ca > ./admin-ca
+
+# Create an yaml file for the ElasticSearch service
+cat << EOF > my_es_route.yaml
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: elasticsearch
+  namespace: openshift-logging
+spec:
+  host:
+  to:
+    kind: Service
+    name: elasticsearch
+  tls:
+    termination: reencrypt
+    caCertificate: |-
+EOF
+cat ./admin-ca | sed -e "s/^/      /" >> my_es_route.yaml
```

Comment:
@richm Can this really only be the admin-ca we generate?  I thought the point was admin CA should only be relevant from the router to ES, and client to router is a well known CA (e.g. provided by Openshift, etc)

Reply:
I think this is the CA cert for the CA that issued the Elasticsearch server cert.  `admin-ca` is slightly misleading - we are assuming that the same CA was used to issue the ES admin cert and the ES http(s) server cert.

Reply:
We could also get the `admin-ca` cert from the secret:
```bash
oc extract secret/elasticsearch --to=. --keys=admin-ca
```

Reply:
I'm not sure that answers my question.  The caCertificate is browser->router.  That should be provided outside the context of logging should it not?  Should it be a BYO ca? Or am I misunderstanding here?

Reply:
caCertificate is optional - https://docs.okd.io/latest/architecture/networking/routes.html#secured-routes "Edge Termination" (which is the same as re-encrypt, in terms of the client/browser facing part)

The key and cert are required - these are the key and cert specifically for the route - not sure what the old openshift-ansible code does for these, if it reuses the Elasticsearch https server cert/key, or generates new server cert/key specifically for the route - https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_logging_elasticsearch/tasks/main.yaml#L508

The caCertificate is optional, depending on if the cert was issued by a well known CA (e.g. something like Verisign which is already in the browsers' CA list), or some sort of corporate CA which is by policy installed on all machines.  In our case, since we are creating the cert/key, we will have to put our CA cert in the caCertificate field, and browsers and other clients will need a copy of this CA cert (or allow insecure connections like `curl -k`).


as far as destinationCACertificate goes:

https://docs.okd.io/latest/architecture/networking/routes.html#secured-routes "Re-encryption Termination"

"destinationCACertificate is required for re-encryption. destinationCACertificate specifies a CA certificate to validate the endpoint certificate, securing the connection from the router to the destination pods. If the service is using a service signing certificate, or the administrator has specified a default CA certificate for the router and the service has a certificate signed by that CA, this field can be omitted.

If the destinationCACertificate field is left empty, the router automatically leverages the certificate authority that is generated for service serving certificates, and is injected into every pod as /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt. This allows new routes that leverage end-to-end encryption without having to generate a certificate for the route. This is useful for custom routers or the F5 router, which might not allow the destinationCACertificate unless the administrator has allowed it."

In our case, we cannot use the default service-ca.crt because our Elasticsearch server cert is custom issued by our custom CA.